modelarmor/create_template_with_labels.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Sample code for creating a new model armor template with labels.
+
+package modelarmor
+
+// [START modelarmor_create_template_with_labels]
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	modelarmor "cloud.google.com/go/modelarmor/apiv1"
+	modelarmorpb "cloud.google.com/go/modelarmor/apiv1/modelarmorpb"
+	"google.golang.org/api/option"
+)
+
+// createModelArmorTemplateWithLabels method creates a
+// new Model Armor template with custom labels.
+//
+// w io.Writer: The writer to use for logging.
+// projectID string: The ID of the Google Cloud project.
+// locationID string: The ID of the Google Cloud location.
+// templateID string: The ID of the template to create.
+// labels map[string]string: A map of custom labels to apply to the template.
+func createModelArmorTemplateWithLabels(w io.Writer, projectID, locationID, templateID string, labels map[string]string) error {
+	ctx := context.Background()
+
+	// Create options for Model Armor client.
+	opts := fmt.Sprintf("modelarmor.%s.rep.googleapis.com:443", locationID)
+	// Create the Model Armor client.
+	client, err := modelarmor.NewClient(ctx,
+		option.WithEndpoint(opts),
```

Comment:
issue: move the call to `option.WithEndpoint()` outside of the client instantiation.

Suggestion:

```go
endpoint := fmt.Sprintf("modelarmor.%s.rep.googleapis.com:443", locationID)
clientOpts = option.WithEndpoint(endpoint)
client, err := modelarmor.NewClient(ctx, clientOpts)
```